src/app/home/home.component.ts

```diff
@@ -45,22 +46,26 @@ export class HomeComponent {
           'currency': self.model.currency,
           'data': tokenData,
         };
-        const apiURL = 'https://donate.keyman.com/api/charge';
+        const apiURL = '/api/charge';
+
+        self.successMessage = '';
+        self.errorMessage = '';
 
         self.http.post(apiURL, postData, {
           headers: new HttpHeaders().set('Content-Type', 'application/json'),
-          observe: 'response', // TODO How do we catch 4xx errors?
+          observe: 'response',
         }).subscribe(
           success => {
             // Handle result
             console.log('POST call successful value returned in body', success);
             self.successMessage = 'The transaction has been completed. Thank you for your donation!';
             self.receiptLink = success.body['receipt_url'];
+            changeDetectorRef.detectChanges();
```

Comment:
This was required for Angular to display the success message, because the event is triggered outside the zone that the Angular framework monitors.